source/arch/metadatastore-format.rst

```diff
@@ -57,31 +57,270 @@ We use the literal ``<time>`` to indicate a client side date/time and
+            "photodiode":0.5}
+  }
+
+and we also want to know *when* we took the data and be able to uniquely identify
+a given event ::
+
+  {
+   "seq_num": 0,
+   "data": {"motor1": 2.5,
+           "photodiode":0.05},
+   "time": 1431698798.9090064,
+   "uid": "82d81f96-2c81-42a4-90e7-9f1c96e5709b"
+  }
+  {
+   "seq_num": 1,
+   "data": {"motor1": 2.4,
+            "photodiode":0.5},
+   "time": 1431698800.8632064
+   "uid": "9912c195-aa5d-4404-b529-823f2a2e30a2"
+  }
+
+However, the data we are measuring is coming from real pieces of hardware which never get
+measured at *exactly* the same time so we also want to store those times ::
+
+
+  {
+   "seq_num": 0,
+    "data": {"motor1": [2.5, 1431698798.9090054],
+            "photodiode": [0.05, 1431698798.9090060]},
+    "time": 1431698798.9090064,
+    "uid": "82d81f96-2c81-42a4-90e7-9f1c96e5709b"
+  }
+  {
+   "seq_num": 1,
+   "data": {"motor1": [2.4, 1431698800.8632061],
+            "photodiode": [0.5, 1431698800.8632062]},
+   "time": 1431698800.8632064
+   "uid": "9912c195-aa5d-4404-b529-823f2a2e30a2"
+  }
+
+For non-scalar data we do not want to store the raw data directly in the events.
+This is both for technical reasons on both the data base side and the collection side
+non-scalar data is stored external to the events and only a reference is stored in
+the documents ::
+
+  {
+   "seq_num": 0,
+    "data": {"cam1": ["c0a687b9-2413-4e18-9bc6-97fe5f049814", 1431698798.9090053],
+             "motor1": [2.5, 1431698798.9090054],
+             "photodiode": [0.05, 1431698798.9090060]},
+    "time": 1431698798.9090064,
+    "uid": "82d81f96-2c81-42a4-90e7-9f1c96e5709b"
+  }
+  {
+   "seq_num": 1,
+   "data": {"cam1": ["4e65245a-88a2-490f-9916-a48c4cf57d68", 1431698800.8632051],
+            "motor1": [2.4, 1431698800.8632061],
+            "photodiode": [ 0.5, 1431698800.8632062]},
+   "time": 1431698800.8632064
+   "uid": "9912c195-aa5d-4404-b529-823f2a2e30a2"
+  }
+
+
+We call these documents "Events" and they capture almost all of the
+scientific information about a measurement.  By bundling all of these
+measurements into a single document the experimenter is communicating that
+this set of measurements are "naturally" associated with each other and are,
+to the experimental time scale that matter, "at the same time".  A stream
+of Events represents a single completely synchronous experiment.
+
+However, Events alone do not tell us:
+
+- where is the external data stored and how do we get it?
+- what hardware was used to take each data
+
+and in general we have no way of knowing ahead of time:
+
+- what the keys in "data" will be
+- what the data type will be
+- if the value is real data or a reference
+- the data shape
+
+For a single data entry the source of the data can be uniquely
+specified with a PV ::
+
+ {
+  "source": "PV:XF23ID{}-RBV"
+ }
+
+the dtype and shape are book keeping::
+
+
+ {
+  "source": "PV:XF23ID{}-RBV",
+  "dtype": "number",
+  "shape": []
```

Comment:
fair point

@dchabot @stuwilkins What do you say to adding `units` to the Event Descriptor? This change would not require a migration.